internal/cmd/instance/update_sharing_instance.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package instance
+
+import (
+	"github.com/Peripli/service-manager-cli/internal/cmd"
+	"github.com/Peripli/service-manager-cli/internal/output"
+	"github.com/Peripli/service-manager-cli/pkg/query"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+type UpdateSharingCmd struct {
+	*cmd.Context
+	instanceName string
+	instanceID   string
+	outputFormat output.Format
+	share        bool
+}
+
+// NewUpdateSharingCmd returns new share/unshare instance command with context
+func NewUpdateSharingCmd(context *cmd.Context, share bool) *UpdateSharingCmd {
+	return &UpdateSharingCmd{Context: context, share: share}
+}
+
+// Prepare returns cobra command
+func (shc *UpdateSharingCmd) Prepare(prepare cmd.PrepareFunc) *cobra.Command {
+	result := &cobra.Command{
+		PreRunE: prepare(shc, shc.Context),
+		RunE:    cmd.RunE(shc),
+	}
+	if shc.share {
+		result.Use = "share-instance [name] --id service-instance-id "
+		result.Short = "Share a service instance"
+		result.Long = `Share a service instance so that it can be consumed from various platforms in your subaccount.
+Instance can be shared only if it was created with the plan that supports instance sharing. For more information, see the documentation of the service whose instance you want to share.`
+	} else {
+		result.Use = "unshare-instance [name] --id service-instance-id "
+		result.Short = "Unshare a service instance"
+		result.Long = `Unshare a service instance to disable its consumption from any but the original platform in which it was created in your subaccount. If an instance you want to unshare has references, an error is returned`
+	}
+	result.Flags().StringVarP(&shc.instanceID, "id", "", "", "Id of the instance. Required in case when there are instances with same name")
```

Comment:
The instance ID. Required if there is more than one instance with the same name.